controllers/managedocs_controller.go

```diff
@@ -177,3 +170,34 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) generateStorageClusterWithParams(configSecret *corev1.Secret) (*ocsv1.StorageCluster, error) {
+	sdata := configSecret.Data
+	requestedStorageCapacity, err := resource.ParseQuantity(string(sdata[storageClassSizeKey]))
+	if err != nil {
+		r.Log.Error(err, "Missing or incorrect parameter in the addonparam secret", "Key", storageClassSizeKey)
+		return nil, err
+	}
+	r.Log.Info("Addon param data", "Storage Cluster Size", requestedStorageCapacity)
+
+	osdSize, _ := resource.ParseQuantity("1Ti")
+
+	// The request must be an exact multiple of the base OSD size
+	rem := requestedStorageCapacity.Value() % osdSize.Value()
+
+	if rem != 0 {
+		err := fmt.Errorf("Storage cluster size must be a multiple of %v", osdSize)
+		return nil, err
+	}
+	count := requestedStorageCapacity.Value() / osdSize.Value()
+	desired := utils.ObjectFromTemplate(templates.StorageClusterTemplate, r.Scheme).(*ocsv1.StorageCluster)
+
+	desired.Spec.StorageDeviceSets[0].Count = int(count)
+	persistentVolume := &desired.Spec.StorageDeviceSets[0].DataPVCTemplate
+	persistentVolume.Spec.Resources = corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			"storage": osdSize,
+		},
+	}
```

Comment:
Why are we bundling this code in the same function as the size treatment? These are 2 separate concerns and should be kept separate. I would suggest creating the template from the outside and passing it to a function that updates the size. 

Currently, we have only one param but with the introduction of more we could just pass the storage cluster to a function for each param which will be responsible to update the template according to the specific param value

Reply:
Done